src/core_modules/capture-core/components/WidgetProfile/DataEntry/ProgramRules/rulesContainer.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+import type { ProgramRulesContainer } from 'capture-core-utils/rulesEngine';
+import { getTrackedEntityAttributeId, getProgramId, getProgramRuleActions } from '../helpers';
+import { getRulesAndVariablesFromProgramIndicators } from '../../../../metaDataMemoryStoreBuilders/programs/getRulesAndVariablesFromIndicators';
+
+const addProgramVariables = (program, programRuleVariables) => {
+    program.programRuleVariables = programRuleVariables
+        .map(programRulesVariable => ({
+            ...programRulesVariable,
+            programId: getProgramId(programRulesVariable),
+            trackedEntityAttributeId: getTrackedEntityAttributeId(programRulesVariable),
+        }));
+};
+
+const addProgramRules = (program, programRules) => {
+    const filteredProgramRules = programRules.map(programRule => ({
+        ...programRule,
+        programId: getProgramId(programRule),
+        programRuleActions: getProgramRuleActions(programRule.programRuleActions),
+    }));
+    const mainRules = filteredProgramRules.filter(rule => !rule.programStageId);
+
+    const rulesByStage = filteredProgramRules
+        .filter(rule => rule.programStageId)
+        .reduce((accRulesByStage, programRule) => {
+            const programStageId = programRule.programStageId || '';
+            accRulesByStage[programStageId] = accRulesByStage[programStageId] || [];
+            accRulesByStage[programStageId].push(programRule);
+            return accRulesByStage;
+        }, {});
+
+    program.programRules = mainRules;
+    Object.keys(rulesByStage).forEach((stageKey) => {
+        const rulesForStage = rulesByStage[stageKey];
+        const programStage = program.getStage(stageKey);
+        if (programStage) {
+            // $FlowFixMe[prop-missing] automated comment
+            programStage.programRules = rulesForStage;
+        }
+    });
```

Comment:
I think the result of this is passed directly into the rulesEngine, if that is the case, we should just create a flat structure of rules here.